crates/telio-dns/src/forward.rs

```diff
@@ -220,9 +220,22 @@ impl Authority for ForwardAuthority {
         telio_log_debug!("forwarding lookup: {} {}", name, rtype);
         let resolve = self.resolver.lookup(name.clone(), rtype).await;
 
-        if resolve.is_err() {
-            telio_log_warn!("DNS name resolution failed with {:?}", resolve);
-        }
+        // Log DNS failures
+        match resolve {
+            // Some errors are caused not really relevant, and happens in a wild often.
+            // For example: no IPs associated with domain especially for AAAA queries
+            //
+            // Avoid logging such errors
+            Err(ref e)
+                if matches!(e.kind(), ResolveErrorKind::NoRecordsFound { .. })
+                    && rtype == RecordType::AAAA => {}
+            Err(ref e)
+                if matches!(e.kind(), ResolveErrorKind::NoRecordsFound { .. })
+                    && rtype == RecordType::SOA => {}
```

Comment:
I think it's worth adding `debug` logs in each case as well as in `Ok()`, makes debugging so much easier